lib/chef/resource/windows_security_policy.rb

```diff
@@ -0,0 +1,111 @@
+#
+# Author:: Ashwini Nehate (<anehate@chef.io>)
+# Author:: Davin Taddeo (<davin@chef.io>)
+# Author:: Jeff Brimager (<jbrimager@chef.io>)
+# Copyright:: 2019-2020, Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "../resource"
+require_relative "../mixin/powershell_out"
+
+class Chef
+  class Resource
+    class WindowsSecurityPolicy < Chef::Resource
+      include Chef::Mixin::PowershellOut
+      resource_name :windows_security_policy
+
+      policy_names = %w(MinimumPasswordAge
+                MaximumPasswordAge
+                MinimumPasswordLength
+                PasswordComplexity
+                PasswordHistorySize
+                LockoutBadCount
+                RequireLogonToChangePassword
+                ForceLogoffWhenHourExpire
+                NewAdministratorName
+                NewGuestName
+                ClearTextPassword
+                LSAAnonymousNameLookup
+                EnableAdminAccount
+                EnableGuestAccount
+                )
+
+      property :id, String, name_property: true, equal_to: policy_names
+      property :secoption, String, required: false, equal_to: policy_names
+      property :secvalue, String, required: true
+      property :sensitive, [true, false], default: true
+     
+      default_action :set
```

Comment:
No need for the default_action. The first one in the list becomes the default

Reply:
Thank you.